### PR TITLE
Fix ee10 path info only

### PR DESCRIPTION
**Pull request by lorban**
Minimalistic fix for #9910

### CODE REVIEW

**Review by gregw — CHANGES_REQUESTED** [review]
This looks about right.... but the test failures are related.   It could be bad test?

**Comment by lorban** [comment]
@gregw I don't think it's a bad test. Schematically, this is what it fails on:

 - `context.setWelcomeFiles(new String[]{"index.html", "index.jsp", "index.htm"})`
 - `GET /context/alt/dir/index.html | assert (status == 200)` (passes)
 - `GET /context/alt/dir/ | assert (status == 200)` (fails)

For the 2nd GET, we have:

`encodedPathInContext == /dir/`
`getWrapped().getHttpURI() == http://local/context/alt/dir/`
`Request.newHttpURIFrom(getWrapped(), encodedPathInContext) == http://local/context/dir/`

So the fix for `testPathInfoOnly` breaks `testWelcomeMultipleDefaultServletsDifferentBases`.

**Comment by gregw** [comment]
@lorban Why is `Request.newHttpURIFrom(getWrapped(), encodedPathInContext) == http://local/context/dir/` rather than `http://local/context/alt/dir/`?    If `isPathInfoOnly` is false, then the encodedPathInContext should be `/alt/dir/`,  if it is true, then `/dir/` must resolve to a resource in the alternative base.

**Comment by lorban** [comment]
@gregw `pathInfoOnly` is set to true, so... could it be a bad test?

**Comment by janbartel** [comment]
The `DefaultServletTest.testWelcomeMultipleDefaultServletsDifferentBases` test that is failing seems like a good test to me. The request sends `http://local/context/other`, which should be matched to the directory `docroot/altroot/dir` because the `pathInfo` on the request is `""`.  As that is a directory, we should send a redirect to the welcome of `http://local/context/other/`, not `http://local/context/`.

**Comment by lorban** [comment]
@janbartel thanks for the clarification.

I've modified the redirect handling as well as the serving of static files to account for pathInfoOnly being true.

**Comment by janbartel** [comment]
A few things I don't understand:
* why in `ResourceService.doGet()` are we passing the `pathInContext` as a separate parameter to `sendWelcome()`, considering that the `Request` is now the `ServletCoreRequest` with its path already modified?
* why do we throw away the `Content` that we found in `DefaultServlet.doGet()` inside `ResourceService.sendWelcome()` when it calls `welcome()`?

It seems to me that we want to use the `pathInfoOnly` to help us select the resource that we want to serve, but if we are going to redirect or send it off to some other servlet to handle then we need to keep the full original path info? @gregw?_

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java [hunk 1]
```diff
@@ -593,13 +593,18 @@ private static class ServletCoreRequest extends Request.Wrapper
             String includedServletPath = (String)request.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
             boolean included = includedServletPath != null;
             if (request.getDispatcherType() == DispatcherType.REQUEST)
-                _uri = getWrapped().getHttpURI();
+                _uri = Request.newHttpURIFrom(getWrapped(), encodedPathInContext);
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java:596]: I'm now cautious about us rewriting the URI.  We should just reconstruct it from servletPath and pathInfo, but not change it for pathInfoOnly.

**Review by gregw — CHANGES_REQUESTED** [review]

**Comment by lorban** [comment]
Superseded by https://github.com/eclipse/jetty.project/pull/9934

**Pull request closed without merging**